tests/unit/common/test_unit_conversions.py

```diff
@@ -48,4 +55,308 @@ def test_multiplication(self, factor1, factor2, expected_product_factor):
+        assert converted_values["mPs2_to_kmPs2"] == 1e-3
+
+    def test_convert_mPs2_knotsPs2(self):
+        unit_converter = UnitConverter(
+            mPs2_to_knotsPs2=ConversionFactors.mPs2_to_knotsPs2,
+            knotsPs2_to_mPs2=ConversionFactors.knotsPs2_to_mPs2,
+        )
+
+        converted_values = unit_converter.convert(mPs2_to_knotsPs2=1.0, knotsPs2_to_mPs2=1.0)
+
+        assert math.isclose(converted_values["mPs2_to_knotsPs2"], 1.94384466, abs_tol=1e-6)
+        assert math.isclose(converted_values["knotsPs2_to_mPs2"], 0.514444, abs_tol=1e-6)
+
+    def test_convert_kg_g(self):
+        unit_convertor = UnitConverter(
+            kg_to_g=ConversionFactors.kg_to_g, g_to_kg=ConversionFactors.g_to_kg
+        )
+
+        converted_values = unit_convertor.convert(kg_to_g=1.0, g_to_kg=1000.0)
+
+        assert converted_values["kg_to_g"] == 1000.0
+        assert converted_values["g_to_kg"] == 1.0
+
+    def test_convert_lb_g(self):
+        unit_convertor = UnitConverter(
+            lb_to_g=ConversionFactors.lb_to_g, g_to_lb=ConversionFactors.g_to_lb
+        )
+
+        converted_values = unit_convertor.convert(lb_to_g=2.5, g_to_lb=1.0)
+
+        assert math.isclose(converted_values["lb_to_g"], 1133.980925, abs_tol=1e-6)
+        assert math.isclose(converted_values["g_to_lb"], 0.00220462, abs_tol=1e-6)
+
+    def test_convert_kg_lb(self):
+        unit_convertor = UnitConverter(
+            kg_to_lb=ConversionFactors.kg_to_lb, lb_to_kg=ConversionFactors.lb_to_kg
+        )
+
+        converted_values = unit_convertor.convert(kg_to_lb=1.5, lb_to_kg=1.0)
+
+        assert math.isclose(converted_values["kg_to_lb"], 3.3069339328, abs_tol=1e-6)
+        assert math.isclose(converted_values["lb_to_kg"], 0.45359237, abs_tol=1e-6)
+
+    @pytest.mark.parametrize(
+        "degrees_to_rad, expected_result",
+        [(0, 0), (90, math.pi / 2), (180, math.pi), (360, 2 * math.pi), (-180, -math.pi)],
+    )
+    def test_convert_degrees_to_rad(self, degrees_to_rad, expected_result):
+        unit_convertor = UnitConverter(
+            degrees_to_rad=ConversionFactors.degrees_to_rad,
+        )
+
+        converted_values = unit_convertor.convert(degrees_to_rad=degrees_to_rad)
+
+        assert math.isclose(converted_values["degrees_to_rad"], expected_result, abs_tol=1e-6)
+
+    @pytest.mark.parametrize(
+        "rad_to_degrees, expected_result",
+        [(0, 0), (math.pi / 2, 90), (math.pi, 180), (2 * math.pi, 360), (-math.pi, -180)],
+    )
+    def test_convert_rad_to_degrees(self, rad_to_degrees, expected_result):
+        unit_convertor = UnitConverter(
+            rad_to_degrees=ConversionFactors.rad_to_degrees,
+        )
+
+        converted_values = unit_convertor.convert(rad_to_degrees=rad_to_degrees)
+
+        assert math.isclose(converted_values["rad_to_degrees"], expected_result, abs_tol=1e-6)
+
+    @pytest.mark.parametrize(
+        "km_to_m, expected_result",
```

Comment:
Testing revealed regular Python lists cause unexpected results during conversion. Swapping `Scalar` to `numpy.typing.ArrayLike` was a plug-and-play process in terms of supporting numpy arrays. More work will have to be done to support Python lists alongside `numpy` arrays.

If this is not acceptable, then changes can be made.

Reply:
More specifically, the lengths of lists seem to be altered during the unit conversion process (alongside getting potentially wrong results)

Reply:
I think we can assume that we would only care about using numpy arrays rather than lists or tuples too. If you can come up with a better way to type other than using `ArrayLike`, then I would be all ears.

Reply:
Given the assumption that we are only using `numpy arrays`, what I have now could be sufficient. I think it could be deployed, but re-requesting review in case more thoughts come up.

Reply:
Taking a look at the numpy documentation, the other alternatives are not as great. This is probably good enough.